### PR TITLE
Add Strengths and Needs API

### PR DESCRIPTION
**Pull request by Thomas-Geraghty**
_note: This is currently written as if SAN had a simple `create` endpoint that just needed the userDetails attribute._

- Added SAN API and configuration
- Added some generic types
- Small other config changes

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/arnscoordinatorapi/integrations/assessment/api/StrengthsAndNeedsApi.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  private val logger = LoggerFactory.getLogger(StrengthsAndNeedsApi::class.java)
+
+  fun createAssessment(body: CreateAssessmentRequest): CreateAssessmentResponse? {
+    return try {
+      sanApiWebClient.post()
+        .uri(apiProperties.endpoints.create)
+        .body(BodyInserters.fromValue(body))
+        .retrieve()
+        .bodyToMono(CreateAssessmentResponse::class.java)
```
Comment (alexw-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/arnscoordinatorapi/integrations/assessment/api/StrengthsAndNeedsApi.kt:25]: Why the WebFlux way here? We're not using it in the other projects.

**Review by devkokov — APPROVED** [review]
`SAN_API_URL` probs needs renaming to `SAN_API_BASE_URL` in the helm files, but other than that LGTM!

**Comment by Thomas-Geraghty** [comment]
> `SAN_API_URL` probs needs renaming to `SAN_API_BASE_URL` in the helm files, but other than that LGTM!

Good spot, resolved in https://github.com/ministryofjustice/hmpps-assess-risks-and-needs-coordinator-api/pull/7/commits/c8a4a38c7bdd5d0bafa06db1ec54c216ffc88a72

File: src/main/kotlin/uk/gov/justice/digital/hmpps/arnscoordinatorapi/integrations/assessment/api/StrengthsAndNeedsApi.kt (hunk 1 above)
Comment (Thomas-Geraghty) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/arnscoordinatorapi/integrations/assessment/api/StrengthsAndNeedsApi.kt:25]: [Just based it off the template](https://github.com/ministryofjustice/hmpps-template-kotlin/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/templatepackagename/service/ExampleApiService.kt#L18-L32)

**Pull request merged**